index.js

```diff
@@ -1,5 +1,40 @@
-export default function getRandomNumber() {
-  console.log(Math.floor(Math.random() * 1000) + 1)
+import http from "http"
+import { usersAPI } from "./src/api/users.js"
+import { hobbiesAPI } from "./src/api/hobbies.js"
+import { returnInvalidPageError } from "./src/api/utils.js"
+
+const API_URL = "/api/users"
+
+const validateURL = (url) => {
+  return !url.includes(API_URL)
 }
 
-getRandomNumber()
+const server = http.createServer((req, res) => {
+  const { url } = req
+  if (validateURL(url)) {
+    returnInvalidPageError(res)
+    return
+  }
+  if (url === API_URL) {
+    usersAPI(req, res)
+  } else {
+    const parsedUrl = url.split("/")
+    const baseUrl = `/${parsedUrl[1]}/${parsedUrl[2]}`
+    if (baseUrl !== API_URL) {
+      returnInvalidPageError(res)
+      return
+    }
+
+    if (parsedUrl.length === 4) {
+      usersAPI(req, res)
+    } else if (parsedUrl.length === 5 && parsedUrl[4] === "hobbies") {
+      hobbiesAPI(req, res)
+    } else {
+      returnInvalidPageError(res)
+    }
+  }
+})
```

Comment:
This works as expected, but you could use regex to ease the mantainability and readability of this handler. Not changes requested.

Reply:
I admit that shamefully I did this because I got lazy and sleepy doing this. Since I have a bit more time, I can try doing regex

Reply:
If have time, try to do it, it will simplify a lot this logic